### ISSUE TITLE
High CPU Usage

### ISSUE THREAD

**Issue by snake3ca** [CocoCR300/flauncher#22]
Flauncher has twice the CPU usage compared to other launcers.

Projectivity: 
   PID USER         PR  NI VIRT  RES  SHR S[%CPU] %MEM     TIME+ ARGS           
  5118 u0_a127      10 -10 1.3G 104M  64M S 20.6   5.3  96:19.53 com.spocky.pro+
  3279 system       -3  -8 138M  16M 8.9M S 10.6   0.8  48:54.12 surfaceflinger

Flauncher(FORK):
   PID USER         PR  NI VIRT  RES  SHR S[%CPU] %MEM     TIME+ ARGS           
  6055 u0_a0        10 -10 1.2G 182M 108M S 41.0   9.3   1:22.82 me.efesser.fla+
  3279 system       -3  -8 138M  16M 8.9M S 13.0   0.8  49:23.53 surfaceflinger

**Comment by Zepmann** [comment]
I can confirm this. Compared to [tsynik/LeanbackLauncher](https://github.com/tsynik/LeanbackLauncher) CPU usage is quite significant.

Note that FLauncher only uses a lot of CPU power when it has focus. Once the device is in "sleep" or another application is opened, FLauncher in idle hardly consumes anything.

Tested on an Nvidia Shield TV Pro (2019).

**Comment by Pentaphon** [comment]
> Compared to [tsynik/LeanbackLauncher](https://github.com/tsynik/LeanbackLauncher) CPU usage is quite significant.

I just compared both and see the same thing. @CocoCR300 this should be first priority.

**Comment by CocoCR300** [comment]
I'll look into this issue now. Thanks for taking the time to measure these metrics.

**Comment by CocoCR300** [comment]
Wasn't expecting something like this, but the moment I disable the "App card highlight animation" setting, the CPU usage drops from up to 60 percent to just 5 percent. I need to confirm if all of you are using this option while monitoring, it is enabled by default.

**Comment by CocoCR300** [comment]
I remember reading this [thread](https://github.com/flutter/flutter/issues/59327) a while ago, trying to see if Flutter could be the problem, because the app itself doesn't do any kind of background work while idle. That issue could be related, now that I see this behavior while toggling the setting I mentioned.

**Comment by Pentaphon** [comment]
> Wasn't expecting something like this, but the moment I disable the "App card highlight animation" setting, the CPU usage drops from up to 60 percent to just 5 percent. I need to confirm if all of you are using this option while monitoring, it is enabled by default.

Can we have an alternative way to highlight cards instead of the animation to avoid this while figuring out how to fix the CPU issue? Perhaps a static highlight?

**Comment by snake3ca** [comment]
> > Wasn't expecting something like this, but the moment I disable the "App card highlight animation" setting, the CPU usage drops from up to 60 percent to just 5 percent. I need to confirm if all of you are using this option while monitoring, it is enabled by default.
> 
> @CocoCR300 Can we have an alternative way to highlight cards instead of the animation to avoid this while figuring out how to fix the CPU issue? Perhaps a static highlight?

Disabling app card highlight animation does lower cpu usage, it would be nice if the animation wasn't so cpu intensive. I agree there should be a static highlight because without the current animation enabled it's hard to see which card is highlighted.

**Comment by Pentaphon** [comment]
> Disabling app card highlight animation does lower cpu usage, it would be nice if the animation wasn't so cpu intensive. I agree there should be a static highlight because without the current animation enabled it's hard to see which card is highlighted.

Agreed. Older people and people with poor eyesight have a harder time seeing it without the animation so I hope the dev introduces a non-animated glow or a high-contrast static border to highlight cards. Those would be good options.
